src/code/sched.c

```diff
@@ -165,48 +230,75 @@ OSScTask* func_800C89D4(SchedContext* sc, OSScTask* task) {
         return NULL;
     }
 
-    if ((sc->pendingSwapBuf2 != NULL ? sc->pendingSwapBuf2->swapBuffer : NULL) == task->framebuffer->fb1) {
+    // If the task's framebuffer is one of the pending swaps or NULL.
+    // In conjunction with the above, these checks are redundant as the pending swap buffers will only be
+    // NULL here, so these could have been simplified to checks for the task's framebuffer being non-NULL.
+
+    if (((sc->pendingSwapBuf2 != NULL) ? sc->pendingSwapBuf2->swapBuffer : NULL) == task->framebuffer->framebuffer) {
         return NULL;
     }
 
-    if ((sc->pendingSwapBuf1 != NULL ? sc->pendingSwapBuf1->swapBuffer : NULL) == task->framebuffer->fb1) {
+    if (((sc->pendingSwapBuf1 != NULL) ? sc->pendingSwapBuf1->swapBuffer : NULL) == task->framebuffer->framebuffer) {
         return NULL;
     }
 
-    if (osViGetCurrentFramebuffer() == (u32*)task->framebuffer->fb1) {
+    // If the task's framebuffer is the current framebuffer, abort
+
+    if (osViGetCurrentFramebuffer() == task->framebuffer->framebuffer) {
         return NULL;
     }
 
     return task;
 }
 
-s32 Sched_Schedule(SchedContext* sc, OSScTask** sp, OSScTask** dp, s32 state) {
+/**
+ * Schedules the next tasks to run on the RSP and RDP
+ *
+ * @param sc Scheduler
+ * @param spTaskOut Next task to run on the RSP
+ * @param dpTaskOut Next task to run on the RDP
+ * @param state Bits containing whether the RSP and RDP are currently in use
+ * @return Bits containing whether the RSP and RDP will be in use after starting the next tasks
+ */
+s32 Sched_Schedule(SchedContext* sc, OSScTask** spTaskOut, OSScTask** dpTaskOut, s32 state) {
     s32 ret = state;
     OSScTask* gfxTask = sc->gfxListHead;
     OSScTask* audioTask = sc->audioListHead;
 
-    if (sc->doAudio && (ret & OS_SC_SP)) {
-        *sp = audioTask;
+    if (sc->doAudio && (state & OS_SC_SP)) {
+        // If we should schedule an audio task and the RSP is available
+        // Mark audio task to enqueue
+        *spTaskOut = audioTask;
+        // RSP no longer available
         ret &= ~OS_SC_SP;
-        sc->doAudio = 0;
+        // Audio task is about to be sent
+        sc->doAudio = false;
+        // Advance task queue
         sc->audioListHead = sc->audioListHead->next;
         if (sc->audioListHead == NULL) {
             sc->audioListTail = NULL;
         }
     } else if (gfxTask != NULL) {
-        if (gfxTask->state & OS_SC_YIELDED || !(gfxTask->flags & OS_SC_NEEDS_RDP)) {
-            if (ret & OS_SC_SP) {
-                *sp = gfxTask;
+        // GFX Task
+        if ((gfxTask->state & OS_SC_YIELDED) || !(gfxTask->flags & OS_SC_NEEDS_RDP)) {
+            // If this is a yielded GFX task, or the RDP is not needed for this GFX task
+            if (state & OS_SC_SP) {
+                // If the RSP is available, enqueue the task and mark the RSP as in use
+                *spTaskOut = gfxTask;
```

Comment:
It's often unclear what conditional comments refer to in chained ifs,
maybe just a line break would help?

```suggestion
        // GFX Task
        if ((gfxTask->state & OS_SC_YIELDED) || !(gfxTask->flags & OS_SC_NEEDS_RDP)) {
            // If this is a yielded GFX task, or the RDP is not needed for this GFX task

            if (state & OS_SC_SP) {
                // If the RSP is available, enqueue the task and mark the RSP as in use
                *spTaskOut = gfxTask;
```